src/global_distrib/mod_global_distrib_hosts_refresher.erl

```diff
@@ -0,0 +1,154 @@
+    gen_server:start_link({local, ?MODULE}, ?MODULE, [RefreshInterval], []).
+
+
+%%--------------------------------------------------------------------
+%% gen_mod callbacks
+%%--------------------------------------------------------------------
+
+-spec start(Host :: ejabberd:lserver(), Opts :: proplists:proplist()) -> any().
+start(Host, Opts) ->
+    mod_global_distrib_utils:start(?MODULE, Host, Opts, fun start/0).
+
+-spec stop(Host :: ejabberd:lserver()) -> any().
+stop(Host) ->
+    mod_global_distrib_utils:stop(?MODULE, Host, fun stop/0).
+
+-spec deps(Host :: ejabberd:server(), Opts :: proplists:proplist()) -> gen_mod:deps_list().
+deps(Host, Opts) ->
+    mod_global_distrib_utils:deps(?MODULE, Host, Opts, fun deps/1).
+
+%%--------------------------------------------------------------------
+%% gen_server callbacks
+%%--------------------------------------------------------------------
+
+init([RefreshInterval]) ->
+    ?DEBUG("refresher starting with interval ~p~n", [RefreshInterval]),
+    schedule_refresh(RefreshInterval),
+    {ok, RefreshInterval}.
+
+handle_call(Request, From, State) ->
+    ?ERROR_MSG("issue=unknown_call request=~p from=~p", [Request, From]),
+    {reply, {error, unknown_request}, State}.
+
+handle_cast(Request, State) ->
+    ?ERROR_MSG("issue=unknown_cast request=~p", [Request]),
+    {noreply, State}.
+
+handle_info(refresh, Interval) ->
+    refresh(),
+    schedule_refresh(Interval),
+    {noreply, Interval}.
+
+terminate(normal, _State) ->
+    ?INFO_MSG("mod_global_distrib_refresher has terminated with reason: normal", []).
+
+
+%%--------------------------------------------------------------------
+%% Helper functions
+%%--------------------------------------------------------------------
+
+
+-spec deps(Opts :: proplists:proplist()) -> gen_mod:deps_list().
+deps(Opts) ->
+    [{mod_global_distrib_mapping, Opts, hard}].
+
+-spec start() -> any().
+start() ->
+    start_outgoing_conns_sup(),
+    Interval = mod_global_distrib_utils:opt(?MODULE, hosts_refresh_interval, default_refresh_interval()),
+    Child = #{
+      id => mod_global_distrib_hosts_refresher,
+      start => {?MODULE, start_link, [Interval]},
+      restart => transient,
+      shutdown => 5000,
+      modules => [mod_global_distrib_hosts_refresher]
+    },
+    {ok, _} = supervisor:start_child(mod_global_distrib_outgoing_conns_sup, Child),
+    ok.
+
+-spec stop() -> any().
+stop() ->
+    stop_outgoing_conns_sup(),
+    ok.
+
+refresh() ->
+    ?DEBUG("Refreshing hosts, checking if there exists a connection pool for all of them", []),
+    Hosts = mod_global_distrib_mapping:hosts(),
+    lists:map(fun maybe_add_host/1, Hosts),
+    ok.
+
+schedule_refresh(Interval) ->
+    erlang:send_after(Interval, self(), refresh).
+
+maybe_add_host(Host) ->
+    case local_host() of
+        Host ->
+            ok;
+        _ ->
+            ?DEBUG("reason=maybe_add_host host=~p local_host=~p", [Host, local_host()]),
```

Comment:
I think adding a new connection pool is `info` worthy (unless it's logged somewhere else).